docs/imagecustomizer/how-to/sysext.md

```diff
@@ -0,0 +1,145 @@
+
+A properly formatted sysext image will typically contain:
+
+| Partition Name                  | Description |
+|----------------------------------|-------------|
+| **Root Filesystem Partition**    | Contains the extension's files (e.g., binaries, libraries, configurations). |
+| **Verity Hash Partition (Optional)** | Stores a Merkle tree hash of the root filesystem, enabling dm-verity for integrity checks. |
+| **Signature Partition (Optional)** | Holds a digital signature verifying the integrity of the hash data. |
+
+
+# Building Sysext Images with mkosi
+mkosi  is the recommended tool for building system extension images. It automates the creation of properly formatted GPT images, including:
+- Partitioning the image (root filesystem, verity, signature).
+- Applying integrity verification using dm-verity.
+- Signing the image (if needed).
+
+For detailed documentation on mkosi commands and configuration, refer to [doc](https://github.com/systemd/mkosi/blob/76377d3838635dd95beae6db393f06258582be2c/mkosi/resources/man/mkosi.1.md)
+
+## 1. Install mkosi
+
+Clone the latest version from GitHub (recommended):
+```  
+git clone https://github.com/systemd/mkosi.git
+cd mkosi
+./bin/mkosi --version
+```
+
+Or install via package manager.
+
+## 2. Define the Image Configuration
+
+1. Create a `mkosi.conf`: defines the build parameters for your sysext image. This configuration file tells mkosi what to include in the image, how to format it, and how to handle verification.
+
+2. Update `sysext.repart.d/` if needed: this directory contains partition definition files (.conf files) that instruct systemd-repart how to create and structure the partitions within the sysext image, including:
+    - Content partition (holds the actual files)
+    - Verity hash partition (used for integrity verification)
+    - Signature partition (optional, contains cryptographic signature)
+
+    Typically, users can create customized sysext.repart.d/ files for certain aspects of the image structure. However, when using Format=sysext - unlike Format=disk, you cannot completely replace these definitions with your own custom repart configurations. **For format=sysext, mkosi is specifically using its own predefined repart definitions located at [mkosi/resources/repart/definitions/sysext.repart.d](https://github.com/systemd/mkosi/tree/e4be13971af896ac6301b78cb045fb4cbe7a2b04/mkosi/resources/repart/definitions/sysext.repart.d). If you need to modify the partition structure (e.g., to change filesystem types or partition sizes), you'll need to edit the .conf files in the mkosi installation directory, which will be in the system paths if installed via package manager or in the cloned repository.**
+
+Example mkosi.conf
+
+```yaml
```

Comment:
Not a YAML file. This is either a toml file or an ini file. (I don't remember which systemd uses.)